rules/DeadCode/Rector/Concat/RemoveUnusedClosureVariableUseRector.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DeadCode\Rector\Concat;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Closure;
+use Rector\PhpParser\Node\BetterNodeFinder;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DeadCode\Rector\Closure\RemoveUnusedClosureVariableUseRector\RemoveUnusedClosureVariableUseRectorTest
+ */
+final class RemoveUnusedClosureVariableUseRector extends AbstractRector
+{
+    public function __construct(
+        private readonly BetterNodeFinder $betterNodeFinder
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Remove unused variable in use() of closure', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+$var = 1;
+
+$closure = function() use ($var) {
+    echo 'Hello World';
+};
+
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+$var = 1;
+$closure = function() {
+    echo 'Hello World';
+};
+
+CODE_SAMPLE
+            ),
+
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Closure::class];
+    }
+
+    /**
+     * @param Closure $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->uses === []) {
+            return null;
+        }
+
+        $hasChanged = false;
+
+        foreach ($node->uses as $key => $useVariable) {
+            $useVariableName = $this->getName($useVariable->var);
+            if (! is_string($useVariableName)) {
+                continue;
+            }
+
+            $isUseUsed = (bool) $this->betterNodeFinder->findVariableOfName($node->stmts, $useVariableName);
```

Comment:
this should verify `compact` use as well, better use `ExprUsedInNodeAnalyzer` for it, I will create new PR for it.

Reply:
I created new PR to skip used via `compact()`

- https://github.com/rectorphp/rector-src/pull/7431